ikarus/controlroutines/pathfollowingfunctions.hh

```diff
@@ -44,6 +44,16 @@ struct SubsidiaryArgs
   Eigen::VectorX<double> dfdDD; ///< The derivative of the subsidiary function with respect to DD.
   double dfdDlambda;            ///< The derivative of the subsidiary function with respect to Dlambda.
   int currentStep;              ///< The current step index in the control routine.
+
+  void reset() {
+    stepSize = 0.0;
```

Comment:
What should this `reset` actually do? Reset means for me that the vectors should also be resized to zero, so resetting to default state.
If this should behave like it is implemented now and the size of the vectors is already known, I suggest renaming it to setZero() instead of reset().

Reply:
You can als pass than the first parameter of the nonlinOP